resources/dw/virtualwarehouse/hive/model_hive_vw.go

```diff
@@ -11,21 +11,151 @@
+	Name                         types.String          `tfsdk:"name"`
+	ImageVersion                 types.String          `tfsdk:"image_version"`
+	GroupSize                    types.Int64           `tfsdk:"group_size"`
+	PlatformJwtAuth              types.Bool            `tfsdk:"platform_jwt_auth"`
+	LdapGroups                   types.List            `tfsdk:"ldap_groups"`
+	EnableSSO                    types.Bool            `tfsdk:"enable_sso"`
+	Compactor                    types.Bool            `tfsdk:"compactor"`
+	JdbcUrl                      types.String          `tfsdk:"jdbc_url"`
+	KerberosJdbcUrl              types.String          `tfsdk:"kerberos_jdbc_url"`
+	HueUrl                       types.String          `tfsdk:"hue_url"`
+	JwtConnectionString          types.String          `tfsdk:"jwt_connection_string"`
+	JwtTokenGenUrl               types.String          `tfsdk:"jwt_token_gen_url"`
+	MinGroupCount                types.Int64           `tfsdk:"min_group_count"`
+	MaxGroupCount                types.Int64           `tfsdk:"max_group_count"`
+	DisableAutoSuspend           types.Bool            `tfsdk:"disable_auto_suspend"`
+	AutoSuspendTimeoutSeconds    types.Int64           `tfsdk:"auto_suspend_timeout_seconds"`
+	ScaleWaitTimeSeconds         types.Int64           `tfsdk:"scale_wait_time_seconds"`
+	Headroom                     types.Int64           `tfsdk:"headroom"`
+	MaxConcurrentIsolatedQueries types.Int64           `tfsdk:"max_concurrent_isolated_queries"`
+	MaxNodesPerIsolatedQuery     types.Int64           `tfsdk:"max_nodes_per_isolated_query"`
+	AwsOptions                   *awsOptions           `tfsdk:"aws_options"`
+	LastUpdated                  types.String          `tfsdk:"last_updated"`
+	Status                       types.String          `tfsdk:"status"`
+	PollingOptions               *utils.PollingOptions `tfsdk:"polling_options"`
 }
 
 func (p *resourceModel) GetPollingOptions() *utils.PollingOptions {
 	return p.PollingOptions
 }
+
+func (p *resourceModel) convertToCreateVwRequest() *models.CreateVwRequest {
+	vwType := models.VwType("hive")
+	return &models.CreateVwRequest{
+		ClusterID:             p.ClusterID.ValueStringPointer(),
+		DbcID:                 p.DatabaseCatalogID.ValueStringPointer(),
+		EbsLLAPSpillGB:        p.getEbsLLAPSpillGB(),
+		ImageVersion:          p.getImageVersion(),
+		Name:                  p.Name.ValueStringPointer(),
+		NodeCount:             utils.Int64To32(p.GroupSize),
+		PlatformJwtAuth:       p.PlatformJwtAuth.ValueBoolPointer(),
+		QueryIsolationOptions: p.getQueryIsolationOptions(),
+		Autoscaling:           p.getAutoscaling(),
+		AvailabilityZone:      p.getAvailabilityZone(),
+		Tags:                  p.getTags(),
+		Config:                p.getServiceConfig(),
+		VwType:                &vwType,
+	}
+}
+
+func (p *resourceModel) getImageVersion() string {
+	if p.ImageVersion.IsNull() || p.ImageVersion.IsUnknown() || p.ImageVersion.String() == "<unknown>" {
```

Comment:
maybe the <unknown> check can be dropped?

Reply:
can you explain what is this `unknown` stuff?

Reply:
I also would request you to reconsider using imageVersion always in your model. If you insert a nil check before using it , you won't have to fight with unknown. As an example from impala

`func (r *impalaResource) createVwRequestFromPlan(plan *resourceModel) *models.CreateVwRequest {
	req := &models.CreateVwRequest{
		Name:      plan.Name.ValueStringPointer(),
		ClusterID: plan.ClusterID.ValueStringPointer(),
		DbcID:     plan.DatabaseCatalogID.ValueStringPointer(),
		VwType:    models.VwTypeImpala.Pointer(),
	}
	if imageVersion := plan.ImageVersion.ValueString(); imageVersion != "" {
		req.ImageVersion = imageVersion
	}
	return req
}`
Here in the last 3 lines, I have set the imageVersion only when it is not nil as such the case where I am not passing image versions have been handled smoothly.

Reply:
Okay, I will add the ValueString() calls in the next patch, it is a better solution